### ISSUE TITLE
Inducing a network delay of 20s leads File System to RO mode.

### ISSUE THREAD

**Issue by dargasudarshan** [openebs/openebs#1410]
## Is this a BUG REPORT or FEATURE REQUEST?
 BUG REPORT
 Choose one: BUG REPORT or FEATURE REQUEST

**What happened**:
1. Deployed Percona application with Liveness probe with the probe performing continuous writes on the volume (In this case, DB writes are performed on percona).
2. Induced network delay of 2000ms(20s) on the controller pod using pumba network emulation tool.
>>kubectl exec pumba-ctrw5 -- pumba netem --interface eth0 --duration 1m delay --time 20000 re2:k8s_pvc-5262a278-37f1-11e8-8ea7-02b983f0a4db-ctrl-con
3. Then observed that percona pod went to Error and subsequently to CrashLoopBackOff state.
On describing the pod observed that File System is in Read Only state, hence the liveness probe failed.

```
ubuntu@kubemaster-01:~$ kubectl get pods
NAME                                                             READY     STATUS    RESTARTS   AGE
maya-apiserver-69f9db69-hs75s                                    1/1       Running   0          3h
openebs-provisioner-77cb47986c-fdmmr                             1/1       Running   2          6d
percona-85b8997987-56bb7                                         0/1       Error     6          8m
pumba-4f7x7                                                      1/1       Running   0          10m
pumba-ft9jv                                                      1/1       Running   0          10m
pumba-tjwp2                                                      1/1       Running   0          10m
pvc-23585dea-37ee-11e8-8ea7-02b983f0a4db-ctrl-5b8596458d-96l98   2/2       Running   0          8m
pvc-23585dea-37ee-11e8-8ea7-02b983f0a4db-rep-cff696bc-4lqkz      1/1       Running   0          8m
pvc-23585dea-37ee-11e8-8ea7-02b983f0a4db-rep-cff696bc-727wr      1/1       Running   0          8m
pvc-23585dea-37ee-11e8-8ea7-02b983f0a4db-rep-cff696bc-swwsh      1/1       Running   0          8m
ubuntu@kubemaster-01:~$ kubectl get pods
NAME                                                             READY     STATUS             RESTARTS   AGE
maya-apiserver-69f9db69-hs75s                                    1/1       Running            0          3h
openebs-provisioner-77cb47986c-fdmmr                             1/1       Running            2          7d
percona-85b8997987-56bb7                                         0/1       CrashLoopBackOff   7          16m
pumba-4f7x7                                                      1/1       Running            0          18m
pumba-ft9jv                                                      1/1       Running            0          18m
pumba-tjwp2                                                      1/1       Running            0          18m
pvc-23585dea-37ee-11e8-8ea7-02b983f0a4db-ctrl-5b8596458d-96l98   2/2       Running            0          16m
pvc-23585dea-37ee-11e8-8ea7-02b983f0a4db-rep-cff696bc-4lqkz      1/1       Running            0          16m
pvc-23585dea-37ee-11e8-8ea7-02b983f0a4db-rep-cff696bc-727wr      1/1       Running            0          16m
pvc-23585dea-37ee-11e8-8ea7-02b983f0a4db-rep-cff696bc-swwsh      1/1       Running            0          16m
ubuntu@kubemaster-01:~$ kubectl describe pod percona-85b8997987-56bb7\
> ^C
ubuntu@kubemaster-01:~$ kubectl describe pod percona-85b8997987-56bb7
Name:           percona-85b8997987-56bb7
Namespace:      default
Node:           kubeminion-02/172.28.128.9
Start Time:     Wed, 04 Apr 2018 09:54:18 +0000
Labels:         name=percona
                pod-template-hash=4164553543
Annotations:    kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicaSet","namespace":"default","name":"percona-85b8997987","uid":"2358806f-37ee-11e8-8ea7-02b983f0a4db"...
Status:         Running
IP:             10.36.0.5
Created By:     ReplicaSet/percona-85b8997987
Controlled By:  ReplicaSet/percona-85b8997987
Containers:
  percona:
    Container ID:  docker://21d23f052d15e15f52ed5df93ca6bf3ec45c90749e0e8a1294017f660396e81b
    Image:         percona
    Image ID:      docker-pullable://percona@sha256:d86cf09f14643a97ef5c1a85c4a5b5636f1a1d58fa44f243175ad620902301d3
    Port:          3306/TCP
    Args:
      --ignore-db-dir
      lost+found
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Wed, 04 Apr 2018 10:10:45 +0000
      Finished:     Wed, 04 Apr 2018 10:10:46 +0000
    Ready:          False
    Restart Count:  8
    Limits:
      cpu:  500m
    Requests:
      cpu:     500m
    Liveness:  exec [bash sql-test.sh] delay=30s timeout=10s period=10s #success=1 #failure=3
    Environment:
      MYSQL_ROOT_PASSWORD:  k8sDem0
    Mounts:
      /sql-test.sh from sqltest-configmap (rw)
      /var/lib/mysql from demo-vol1 (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-pc7wn (ro)
Conditions:
  Type           Status
  Initialized    True
  Ready          False
  PodScheduled   True
Volumes:
  demo-vol1:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  demo-vol1-claim
    ReadOnly:   false
  sqltest-configmap:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      sqltest
    Optional:  false
  default-token-pc7wn:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-pc7wn
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.alpha.kubernetes.io/notReady:NoExecute for 300s
                 node.alpha.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason                 Age                From                    Message
  ----     ------                 ----               ----                    -------
  Warning  FailedScheduling       16m (x2 over 16m)  default-scheduler       PersistentVolumeClaim is not bound: "demo-vol1-claim" (repeated 4 times)
  Normal   Scheduled              16m                default-scheduler       Successfully assigned percona-85b8997987-56bb7 to kubeminion-02
  Normal   SuccessfulMountVolume  16m                kubelet, kubeminion-02  MountVolume.SetUp succeeded for volume "sqltest-configmap"
  Normal   SuccessfulMountVolume  16m                kubelet, kubeminion-02  MountVolume.SetUp succeeded for volume "default-token-pc7wn"
  Normal   SuccessfulMountVolume  16m                kubelet, kubeminion-02  MountVolume.SetUp succeeded for volume "pvc-23585dea-37ee-11e8-8ea7-02b983f0a4db"
  Warning  Unhealthy              12m (x3 over 12m)  kubelet, kubeminion-02  Liveness probe failed: mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1007 (HY000) at line 1: Can't create database 'Inventory'; database exists
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1146 (42S02) at line 1: Table 'Inventory.Hardware' doesn't exist
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1146 (42S02) at line 1: Table 'Inventory.Hardware' doesn't exist
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 6 (HY000) at line 1: Error on delete of './Inventory//db.opt' (Errcode: 30 - Read-only file system)
  Normal   Killing     12m                kubelet, kubeminion-02  Killing container with id docker://percona:Container failed liveness probe.. Container will be killed and recreated.
  Normal   Pulled      11m (x3 over 16m)  kubelet, kubeminion-02  Successfully pulled image "percona"
  Normal   Started     11m (x3 over 16m)  kubelet, kubeminion-02  Started container
  Normal   Created     11m (x3 over 16m)  kubelet, kubeminion-02  Created container
  Warning  BackOff     11m (x3 over 11m)  kubelet, kubeminion-02  Back-off restarting failed container
  Normal   Pulling     11m (x4 over 16m)  kubelet, kubeminion-02  pulling image "percona"
  Warning  FailedSync  1m (x48 over 11m)  kubelet, kubeminion-02  Error syncing pod
```
**What you expected to happen**:
Network delay of 20s should be sustained.

**How to reproduce it (as minimally and precisely as possible)**:
^^Above
**Anything else we need to know?**:

**Comment by muratkars** [comment]
@dargasudarshan Did #1415 solve this issue? If yes, can we close this one?

**Comment by dargasudarshan** [comment]
@payes Attaching the latest logs with 0.6.0-RC2.
[network_failure_test.log](https://github.com/openebs/openebs/files/2113847/network_failure_test.log)

**Comment by dargasudarshan** [comment]
This issue is resolved in 0.7.0
